adapters/repos/db/index.go

```diff
@@ -424,10 +414,24 @@ func (i *Index) IterateObjects(ctx context.Context, cb func(index *Index, shard
 }
 
 func (i *Index) ForEachShard(f func(name string, shard ShardLike) error) error {
+	i.closeLock.RLock()
```

Comment:
Should we factor that check in a single function we can call at the beginning of (almost) all functions ?

Something like
```
if err := i.isClosed(); err != nil {
 return err
}
```

Reply:
that won't help, the intention was to keep the lock during the entire iteration. Nevertheless I reverted this change because this protection would involve changes in the backup process. So that may be addressed in a different PR